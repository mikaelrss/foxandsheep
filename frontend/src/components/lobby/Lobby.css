:local(.roomList) > div {
  height: 50px;
  width: 100%;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border: 5px solid #69e6ff;
  margin-bottom: 10px; }
  :local(.roomList) > div:hover {
    background-color: #e6e6e6; }
  :local(.roomList) > div a {
    width: 100%;
    height: 100%;
    text-align: center;
    line-height: 40px;
    text-decoration: none;
    color: #282828;
    font-weight: bold; }
