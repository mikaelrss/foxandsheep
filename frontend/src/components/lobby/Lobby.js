// @flow

import React, { Component } from 'react';
import { withRouter, Switch, Route, Link } from 'react-router-dom';

import type { Socket } from 'socket.io-client';

import Game from '../game/Game';
import socketClient from 'socket.io-client';
import style from './Game.css';

type Props = {};

type State = {
  socket: Socket,
  roomNames: Array<string>,
};

class Lobby extends Component<Props, State> {
  constructor(props) {
    super(props);
    const socket = socketClient('http://localhost:4000');

    socket.on('roomNames', this.initRoomNames);
    socket.on('roomsUpdated', this.updateRoomNames);

    this.state = {
      socket,
      roomNames: [],
    };
  }

  initRoomNames = payload => {
    console.log('TESTESSSs', payload);
    this.setState({
      roomNames: payload.roomNames,
    });
  };

  updateRoomNames = payload => {
    this.setState({
      roomNames: payload.roomNames,
    });
  };

  createRoom = () => {
    this.state.socket.emit('createRoom');
    this.props.history.push({ pathname: '/game/', state: { createRoom: true } });
  };

  render() {
    console.log(this.state);
    const joinedRoom = this.props.location.pathname.includes('game');
    return (
      <div>
        <Switch>
          <Route path="/game" render={() => <Game cellSize={50} socket={this.state.socket} />} />
        </Switch>
        {!joinedRoom && (
          <div>
            <h3>Rooms</h3>
            <div className={style.roomList}>
              {this.state.roomNames.map(roomName => (
                <div key={roomName}>
                  <Link to="/game">{roomName}</Link>
                </div>
              ))}
            </div>
            <button onClick={this.createRoom}>Create room</button>
          </div>
        )}
      </div>
    );
  }
}

export default withRouter(Lobby);
