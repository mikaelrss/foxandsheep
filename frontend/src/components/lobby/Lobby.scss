@import '../../../styles/Colors';

:local(.roomList)  {
  > div {
    height: 50px;
    width: 100%;
    display: inline-flex;
    justify-content: center;
    align-items: center;

    border: 5px solid $blue;
    margin-bottom: 10px;

    &:hover {
      background-color: darken(white, 10);
    }

    a {
      width: 100%;
      height: 100%;
      text-align: center;
      line-height: 40px;

      text-decoration: none;
      color: #282828;
      font-weight: bold;
    }
  }
}
