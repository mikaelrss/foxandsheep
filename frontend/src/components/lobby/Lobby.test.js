import React from 'react';
import { shallow } from 'enzyme';

import Lobby from './Lobby';

describe('', () => {
  it('shall match snapshot', () => {
    const wrapper = shallow(<Lobby />);

    expect(wrapper).toMatchSnapshot();
  });

  it('shall fail', () => {
    expect(false).toBe(true);
  });
});
