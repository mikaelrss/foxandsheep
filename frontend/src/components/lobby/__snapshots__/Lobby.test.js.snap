// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shall match snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
